helix-view/src/document.rs

```diff
@@ -507,6 +509,38 @@ impl Document {
     pub fn versioned_identifier(&self) -> lsp::VersionedTextDocumentIdentifier {
         lsp::VersionedTextDocumentIdentifier::new(self.url().unwrap(), self.version)
     }
+
+    pub fn diagnostics(&self) -> &[Diagnostic] {
+        &self.diagnostics
+    }
+
+    pub fn set_diagnostics(&mut self, diagnostics: Vec<Diagnostic>) {
+        self.diagnostics = diagnostics;
+        self.current_diagnostic = 0;
+    }
+
+    pub fn next_diagnostic(&mut self) -> Option<&Diagnostic> {
+        let idx = self.current_diagnostic.wrapping_add(1);
+        let idx = if idx < self.diagnostics.len() { idx } else { 0 };
```

Comment:
Is doing modulo bad here? Why not `% self.diagnostics.len()`?

Reply:
Should be modulo, but I still think this should use the current cursor position instead (next = next diagnostic after cursor).